### ISSUE TITLE
Orientdb getVertices() returning Vertex for any label present in database after vertex indexed  with unique name(String) property

### ISSUE THREAD

**Issue by prakashpandey** [orientechnologies/orientdb#6859]
## 
- v2.2 SNAPSHOT[ ] - .10
## Operating System
- Linux
## Expected behavior and actual behavior

Orientdb getVertices(lable, [] iKey, [] iValue)) returing Vertex for any label present in database after I index my vertex by unique name(String) property
## Steps to reproduce the problem

```
`public class App 
{
    public static void main( String[] args )
    {
        String DB_NAME = "WonderLand";
        OrientGraph graph = Service.getOrCreateGraphInstance(DB_NAME);


        OrientVertexType ovt = graph.createVertexType("Person");
        OrientVertexType ovt1 = graph.createVertexType("Relation");

        graph.createKeyIndex("uid", Vertex.class, new Parameter("type", "UNIQUE"));


    OrientVertex v = graph.addVertex("class:Person");

        v.setProperties("uid", "ram");
        v.setProperties("event", "Login");



        Iterable<Vertex> it = graph.getVertices("Person", new String[]{"uid"}, new String[]{"ram"});
        System.out.println("vertex Person->> " + it.iterator().next());

        Iterable<Vertex> it2 = graph.getVertices("Relation", new String[]{"uid"}, new String[]{"ram"});
        System.out.println("vertex Relation->> " + it2.iterator().next());

       }
}
`
```
## Result with Indexing :
- vertex Person->> v(Person)[#-1:-2]
- vertex Relation->> v(Person)[#-1:-2]
## Result without Indexing :
- vertex Person->> v(Person)[#-1:-2]

kindly have a look at this [http://stackoverflow.com/questions/39999533/orientdb-getvertices-returing-vertex-for-any-label-present-in-database](stack overflow link) for more detail.

**Comment by tglman** [comment]
hi @prakashpandey.

with the `graph.createKeyIndex` you are creating a index in the whole database, use instead:
```
    OrientVertexType vertexType = graph.createVertexType("Person");
    OrientVertexProperty prop = vertexType.createProperty("uid", OType.STRING);
    prop.createIndex(INDEX_TYPE.UNIQUE);
```

hope this help

**Comment by tglman** [comment]
Hi @prakashpandey,

Yes anyway there is a discrepancy on the getVertices behaviour that should be fixed, thanks for reporting.

Bye

**Comment by prakashpandey** [comment]
Thanks, I will try v2.2.13.

Hopefully will get resolved in that version.

**Comment by tglman** [comment]
hi,

fixed also this incongruency, will be released in 2.2.14, closing
